batching_kafka_consumer/__init__.py

```diff
@@ -218,7 +220,13 @@ def _handle_message(self, msg):
                 raise
         else:
             if result is not None:
-                self.batch.append(result)
+                self.__batch_results.append(result)
+        finally:
+            duration = (time.time() - start) * 1000
+            self.__batch_messages_processed_count += 1
+            self.__batch_messages_processed_duration_ms += duration
+            if self.metrics:
+                self.metrics.timing('process_message', duration)
```

Comment:
This will be relatively high volume — I can remove it, if need be (or we can sample it) but I think this is useful to have for comparing to the normalized time.

Reply:
This also doesn't distinguish between these three cases which might have different latency profiles:

- successful process yielding a result,
- successful process yielding no result,
- failed process

Reply:
I would advise to add the sampling rate as a constructor parameter, it is a one line change. See the Stats file in cdc. That way it would be trivial to change it as a config parameter

Reply:
My only issue with setting this as a global sampling rate is that this metric is going to be sent several orders of magnitude more frequently (literally) than the others, it seems like it would make a lot more sense to just apply the sampling rate on a per-metric basis here.

Reply:
yes, I was talking about a per metric sampling rate in that it is a parameter of the "timing" method call.

https://github.com/getsentry/cdc/blob/master/cdc/utils/stats.py#L67